app/src/store/bootstrap.ts

```diff
@@ -0,0 +1,49 @@
+import { createAsyncThunk } from '@reduxjs/toolkit';
+import { type Dispatch } from './app';
+import { checkLoginStatus } from './user';
+import { getAccessToken } from './voice/accessToken';
+import { register } from './voice/registration';
+
+type BootstrapAppRejectValue =
+  | {
+      reason: 'CHECK_LOGIN_STATUS_REJECTED';
+    }
+  | {
+      reason: 'GET_ACCESS_TOKEN_REJECTED';
+    }
+  | {
+      reason: 'REGISTER_REJECTED';
+    };
+
+type BootstrapAppFulfillValue = 'NOT_LOGGED_IN' | 'LOGGED_IN';
+
+export const bootstrapApp = createAsyncThunk<
```

Comment:
Why do we need to "bootstrapApp" now? What was wrong with the previous login implementation?

Reply:
Having a bootstrapping action allows us to place more complex logic into our app and have it start somewhere outside of a component. I issue this action in the `App.tsx` file before even rendering the app, and so by the time the rendering is done we can already skip the login screen if the user was already logged in.

Reply:
The previous login implementation is also good, but I think this is a good improvement.